src/geo/map/cartodb-layer.js

```diff
@@ -137,10 +138,29 @@ var CartoDBLayer = LayerModelBase.extend({
 
   update: function (attrs) {
     if (attrs.source) {
-      throw new Error('"source" must be set via setSource');
+      console.warn('Deprectaed: Use ".setSource" to update a layar\'s source instead the update method');
+      attrs.source = CartoDBLayer.getLayerSourceFromAttrs(attrs, this._vis.analysis);
     }
     LayerModelBase.prototype.update.apply(this, arguments);
   }
 });
 
+/**
+  * Return the source analysis node from given attrs object.
+  */
+CartoDBLayer.getLayerSourceFromAttrs = function (attrs, analysis) {
```

Comment:
FYI, Backbone gives you a way to define static methods/properties:

```
var MyModel = Backbone.Model.extend({
  initialize: function () { ... },
  ...
}, {
  myStaticProperty: 1,
  myStaticMethod: function () { ... }
  ...
});
```

More info: http://backbonejs.org/#Model-extend